src/zenml/steps/utils.py

```diff
@@ -318,6 +318,10 @@ def resolve_input_artifact(
         Returns:
             Return the output of `handle_input()` of selected materializer.
         """
+        # Make an exception for BaseArtifact, return artifact
+        if data_type is BaseArtifact:
```

Comment:
Do you think it makes sense to change both checks in this file to subclass checks using `issubclass(...)` to also allow other artifact types?

Reply:
I implemented this now :-)